File: Ruby/spec/parrot_spec.rb
require_relative '../lib/parrot'

describe "Parrots" do
  it "gets speed of a European parrot" do
    parrot = EuropeanParrot.new()
    expect(parrot.speed).to eq(12.0)
  end

  it "gets speed of an African parrot with one coconut" do
    parrot = AfricanParrot.new(1)
    expect(parrot.speed).to eq(3.0)
  end

  it "gets speed of an African parrot with two coconuts" do
    parrot = AfricanParrot.new(2)
    expect(parrot.speed).to eq(0.0)
  end

  it "gets speed of an African parrot with no coconuts" do
    parrot = AfricanParrot.new(0)
    expect(parrot.speed).to eq(12.0)
  end

  it "gets speed of a nailed Norwegian blue parrot" do
    parrot = NorwegianBlueParrot.new(:norwegian_blue_parrot, 0, 1.5, true)
    expect(parrot.speed).to eq(0.0)
  end

  it "gets speed of a not nailed Norwegian blue parrot" do
    parrot = NorwegianBlueParrot.new(:norwegian_blue_parrot, 0, 1.5, false)
    expect(parrot.speed).to eq(18.0)
  end

  it "gets speed of a not nailed Norwegian blue parrot with high voltage" do
    parrot = NorwegianBlueParrot.new(:norwegian_blue_parrot, 0, 4, false)
    expect(parrot.speed).to eq(24.0)
  end
end

File: JavaScript/src/__tests__/parrot.spec.js
import { PARROT_TYPES, EuropeanParrot, AfricanParrot, NorwegianBlueParrot } from '../parrot';

describe("Parrot", function () {
    test("get speed of european parrot", function () {
        const parrot = new EuropeanParrot();

        expect(parrot.getSpeed()).toBe(12);
    });

    test("get speed of african parrot with one coconut", function () {
        const parrot = new AfricanParrot(1);

        expect(parrot.getSpeed()).toBe(3);
    });

    test("get speed of african parrot with two coconuts", function () {
        const parrot = new AfricanParrot(2);

        expect(parrot.getSpeed()).toBe(0);
    });

    test("get speed of african parrot with no coconuts", function () {
        const parrot = new AfricanParrot(0);

        expect(parrot.getSpeed()).toBe(12);
    });

    test("get speed norwegian blue parrot nailed", function () {
        const parrot = new NorwegianBlueParrot(PARROT_TYPES.NORWEGIAN_BLUE, 0, 1.5, true);

        expect(parrot.getSpeed()).toBe(0);
    });

    test("get speed norwegian blue parrot not nailed", function () {
        const parrot = new NorwegianBlueParrot(PARROT_TYPES.NORWEGIAN_BLUE, 0, 1.5, false);

        expect(parrot.getSpeed()).toBe(18);
    });

    test("get speed norwegian blue parrot not nailed high voltage", function () {
        const parrot = new NorwegianBlueParrot(PARROT_TYPES.NORWEGIAN_BLUE, 0, 4, false);

        expect(parrot.getSpeed()).toBe(24);
    });
});
